src/app/contribute/guidelines/component-guidelines/index.html

```diff
@@ -0,0 +1,269 @@
+    </li>
+    <li>
+      <b>All new components and visual changes to existing components must be accompanied by visual regression tests.</b> This ensures that future changes to CSS or markup will not cause components to render in an unexpected manner. Visual tests consist of three parts: an HTML template for the component to test, a TypeScript file for the component to test, and the actual file that runs the test using webdriver.io and our custom screenshot functions. You can see examples of each part of the visual test process at:
+    </li>
+    <ul>
+      <li>
+        <a stacheRouterLink="https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.html">https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.html</a>
+      </li>
+      <li>
+        <a stacheRouterLink="https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.ts">https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.ts</a>
+      </li>
+      <li>
+        <a stacheRouterLink="https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert.visual-spec.ts">https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert.visual-spec.ts</a>
+      </li>
+    </ul>
+    <li>
+      <b>All new components and changes to existing components must have passing <a stacheRouterLink="/learn/accessibility/test">accessibility tests</a>.</b> This includes automated tests to run during visual regression tests as well as code review and manual keyboard testing.
+    </li>
+    <li>
+      <b>All new components must include demos to accompany the reference documentation in the <a stacheRouterLink="/components">Components section</a> of the SKY UX website.</b> Demos must include all or most features available to the components and examples of all inputs and outputs. In the <a stacheRouterLink="https://github.com/blackbaud/skyux2"></a>SKY UX repo, you create subfolders for your components in the <stache-code>src/demos</stache-code> folder and add TypeScript and HTML files for the demos. You create <stache-code>{{ '{' }}componentName}-demo.component.html</stache-code> files to provide HTML templates for demos, <stache-code>{{ '{' }}componentName}-demo.component.ts</stache-code> files to handle the logic, and <stache-code>index.ts</stache-code> files to export demos. For examples of existing demos, serve the the <a stacheRouterLink="https://github.com/blackbaud/skyux2">SKY UX repo</a> or view the <a stacheRouterLink="/components">components documentation</a>. We recommend that you also <a stacheRouterLink="." fragment="write-documentation">create the reference documentation for your components</a> in a separate pull request in the <a stacheRouterLink="https://github.com/blackbaud/skyux2-docs">SKY UX docs repo</a>.
```

Comment:
Use `href` instead of `stacheRouterLink` for external links.  This sentence may need to be rewritten completely though.  Can you comment on where demos should live @Blackbaud-SteveBrush?

Reply:
Demos currently live in the `skyux2` repo. There are talks about moving those to the docs repo, however.